### PR TITLE
Validate required status for attachments

### PR DESCRIPTION
**Pull request by wolverian**

### CODE REVIEW

**Comment by elahti** [comment]
Previously, when we had the possibility to go with a design where an application could be submitted even without providing required attachments, the feature listing all the missing columns was changed to not show missing required attachments.

screen shot 2017-03-15 at 09 39 56

As a reference, in the picture above is the submit button being grayed out due to missing required answers. There are 3 missing required answers. The missing required attachment is not rendered there.

So in a nutshell, all answers of type attachment are filtered out while composing that list. That functionality just needs to be removed.

File: src/cljc/ataru/hakija/application_validators.cljc [hunk 1]
```diff
@@ -9,7 +9,9 @@
 
 (defn ^:private required?
   [value]
-  (not (clojure.string/blank? value)))
+  (cond
```
Comment (elahti) [inline src/cljc/ataru/hakija/application_validators.cljc]: An `if` could be used here as well?

Comment (wolverian) [inline src/cljc/ataru/hakija/application_validators.cljc]: Yeah, I BDUFfed in case of more branches in the future.

**Comment by wolverian** [comment]
Fixed the submit button popup thingy. It shows attachments too now.

File: src/cljc/ataru/hakija/application_validators.cljc (hunk 1 above)
Comment (wolverian) [inline src/cljc/ataru/hakija/application_validators.cljc]: Fixed, thanks.

**Comment by wolverian** [comment]
> maybe we could have a function that digs up a form's component from the app state when it is provided the component's id as a parameter?

Sure, but I'd do that in a separate PR. (In other words, I probably wouldn't do it at all, because it doesn't feel important enough.)
